text/0000-llm-text-docs.md

```diff
@@ -0,0 +1,125 @@
+# Feature Name: rustdoc_text_format
+
+Start Date: 2024-12-30
+RFC PR: (leave this empty)
+Rust Issue: (leave this empty)
+
+# Summary
+Add a new rustdoc output format that generates a simplified, AI-friendly version of the crate's public API surface. This format excludes private items and function implementations while preserving documentation, type signatures, and the module structure.
+
+# Motivation
+As artificial intelligence becomes increasingly important in software development, there's a growing need for machine-readable documentation that can help AI systems quickly understand crate structure and capabilities. Current documentation formats are either:
+
+1. Too verbose (full source code)
+2. Too sparse (generated HTML docs)
+3. Not machine-optimized (markdown/text documentation)
+
+This proposal aims to create an intermediate format that maintains the essential structure and documentation while removing implementation details that aren't necessary for understanding the public API.
+
+# Guide-level explanation
+The new format can be generated using a new rustdoc flag:
+
+```bash
+cargo rustdoc --output-format=text
+```
+
+This will generate a .txt file containing the crate's public API surface, structured similarly to the source code but with the following modifications:
+
+- All private items (functions, structs, fields, etc.) are excluded
+- Function bodies are omitted
+- Documentation comments are preserved
+- Type signatures and trait bounds are preserved
+- Module structure is maintained
+- Macros are included with their documentation but not their implementation
+
+Example output:
+
+```rust
+/// A collection type that stores elements in sorted order
+pub struct BTreeMap<K, V> 
+where 
+    K: Ord
+{
+    /// The comparison function used to maintain ordering
+    pub comparator: Option<Box<dyn Fn(&K, &K) -> Ordering>>,
+}
+
+impl<K: Ord, V> BTreeMap<K, V> {
+    /// Creates an empty BTreeMap
+    /// 
+    /// # Examples
+    /// ```
+    /// use std::collections::BTreeMap;
+    /// let map: BTreeMap<i32, &str> = BTreeMap::new();
+    /// ```
+    pub fn new() -> Self
```

Comment:
```suggestion
    pub fn new() -> Self;
```

It would be much more useful if the output can at least be used by a standard parser.

(Alternatively reuse `-Z unpretty=everybody_loops` output

```suggestion
    pub fn new() -> Self { loop {} }
```

)